packages/material-ui-based-components/src/components/TextField.tsx

```diff
@@ -1,10 +1,78 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import { withRuleEngine } from '../shared/withRuleEngine';
 import { PROPS } from '../utils/types';
+import { FormControl, FormHelperText, IconButton, InputAdornment, InputLabel, OutlinedInput } from '@mui/material';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
 
-const TextField = (props: PROPS)=>{
-  console.log('====props', props);
-  return <div></div>;
+const TextFieldComponent = (props: PROPS) => {
+  const { isError, placeholder, name, required, label, format, errorMessage,
+    description, enabled, readOnly, maxLength, minLength, pattern, id, value
+  } = props;
+
+  const changeHandler = useCallback((event: any) => {
+    props.dispatchChange(event.target.value);
+  }, [props.dispatchChange]);
+
+  const blurHandler = useCallback((event: any) => {
+    props.dispatchBlur(event.target.value);
+  }, [props.dispatchBlur]);
+
+  const focusHandler = useCallback((event: any) => {
+    props.dispatchFocus(event.target.value);
+  }, [props.dispatchFocus]);
+
+  const isPassword = props.fieldType === 'password-input';
+  const [showPassword, setShowPassword] = React.useState(true);
+
+  const handleClickShowPassword = () => setShowPassword((show) => !show);
+
+  const getPasswordIcon = () => {
+    return (
+      <InputAdornment position="end">
+        <IconButton
+          name="passwordButton"
+          onClick={handleClickShowPassword}
+          edge="end"
+          title="passwordButton"
+        >
+          {showPassword ? <Visibility /> : <VisibilityOff />}
+        </IconButton>
+      </InputAdornment>
+    );
+  };
+
+  return (
+    <FormControl
+      variant="outlined"
```

Comment:
variant should be dynamic. It will come from the props.properties['afs:layout'].variant and default value of this outlined

Reply:
Added variant